pkgs/by-name/s7/s7/package.nix

```diff
@@ -0,0 +1,79 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  notcurses,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "s7";
+  version = "0-unstable-2024-10-29";
+
+  src = fetchFromGitLab {
+    domain = "cm-gitlab.stanford.edu";
+    owner = "bil";
+    repo = "s7";
+    rev = "15ff6156a0c836bda702cbf816c370daf9c83910";
+    hash = "sha256-anH9gPhAbxvv3XKb9/zEk4U8pRoXwVApWGu0S97oqbk=";
+  };
+
+  nativeBuildInputs = [
+    notcurses
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    gcc s7.c \
+        -o s7-nrepl \
+        -O2 \
+        -I. \
+        -Wl,-export-dynamic \
+        -lm \
+        -ldl \
+        -DWITH_MAIN \
+        -DWITH_NOTCURSES \
+        -lnotcurses-core
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    dst_bin=$out/bin
+    dst_lib=$out/lib/s7
+    mkdir -p $dst_bin
+    mkdir -p $dst_lib
+
+    cp -prv -t $dst_bin \
+        s7-nrepl
+    ln -s $dst_bin/s7-nrepl $dst_bin/s7
+
+    cp -prv -t $dst_lib \
+        s7.c \
+        s7.h
```

Comment:
So, in what usecase it is better to have `s7` package install the sources than to have them taken from `s7.src`?